### PR TITLE
feat: Allow <Pin> glyphs to be passed as children (close #98)

### LINKED ISSUE #98: [Feat] Allow `<Pin>` glyphs to be passed as children
### Target Use Case

Currently, the only way pass a glyph thats an `HTMLElement` to a `<Pin>` element (as far as i can tell) is to create the element inline in a react component and pass that to the pin like in the following example:

```tsx
export default function MyComponent() {
  const glyphIcon = document.createElement('img');
  glyphIcon.src = "/custom-icon.svg";

  return (
    <>
      <AdvancedMarker>
        <Pin glyph={glyphIcon}/>
      </AdvancedMarker>
    </>
  )
}
```

This also causes unexpected issues when trying to render multiple markers using `Array.prototype.map()`, where defining the element just once in the component will only render it out on the last `AdvancedMarker`, since the native DOM and reacts virtual DOM have some conflicts when using `document.createElement()`. The workaround is to move the `document.createElement()` part down _into_ map, but still not a clean way to interact with react.

Instead, I think it would be much nicer if we could pass a glyph via react `children`, similar to how `<MapControl>` currently functions, which renders the children to appropriate `HTMLElements` to be used with the google maps api. As this is a _react_ wrapper, removing the need to manually create DOM elements directly would be inline with what this library is trying to achieve.

This would increase code maintainability, decrease weird "non-react-like" code and decrease bug encounters when trying to interface with native DOM Elements and reacts virtual DOM when using this library (as well as just making the API easier to use).

### Proposal

Ideally, when trying to pass a glyph thats an HTMLElement to a `<Pin>`, we could do something like this:
```tsx
<AdvancedMarker>
  <Pin background={...}>
    <img src="/icon.svg" />
  </Pin>
</AdvancedMarker>
```

Where `<Pin>` interprets `children` that get passed as glyphs of type `Element` (according to types [here](https://developers.google.com/maps/documentation/javascript/reference/advanced-markers#PinElementOptions)) and gets rendered out appropriately.

### PR DESCRIPTION
**Pull request by benschlegel**
This PR allows the `glyph` property of <Pin> to be passed via JSX children. Functionality is implemented like discussed in #98.

It uses a div container for the portal similar to <MapControl> (where the memo'd div wasn't added as a dependency to the useEffect, I assume that was intentional and also implemented it in this way for <Pin>).

I don't think theres any benefit to using a more detailed type for props to differentiate if children were passed or not, as those are inaccurate due to typescript/tsx limitations unfortunately, so I just wrapped the existing type with `PropsWithChildren`.

I took the code style of the other components as reference, but if there's anything to change im happy to do so :)

### CODE REVIEW

**Review by usefulthink — COMMENTED** [review]
Excellent work, thanks a lot!
Two minor changes, and we can merge it.

Did you see any problems with the extra div-element wrapping the content of the glyph or does it work as expected?

File: examples/markers-and-infowindows/src/app.tsx [hunk 1]
```diff
@@ -43,6 +43,19 @@ const App = () => {
...
         </AdvancedMarker>
 
+        {/* advanced marker with html pin glyph */}
+        <AdvancedMarker
+          position={{lat: 15, lng: 20}}
+          title={'AdvancedMarker with customized pin.'}>
+          <Pin background={'#22ccff'} borderColor={'#1e89a1'} scale={1.4}>
+            {/* child gets rendered as 'glyph' element of pin */}
+            <img
+              src="https://www.svgrepo.com/show/522904/info-circle.svg"
```
[the remark this answers is not in the document]
Comment (benschlegel) [reply examples/markers-and-infowindows/src/app.tsx:53]: Okay, will change it, just did this as a suggestion as I wasn't sure about how you want to handle it in this repo. Any preference for what svg to choose for the example? I just picked a copyright free one from svgrepo.

**Comment by benschlegel** [comment]
Tested with a few different configurations but did not see a problem with the extra div, I'm not sure what edge cases this could cause a problem in but I didn't run into them

File: src/components/pin.tsx [hunk 2]
```diff
@@ -24,15 +33,35 @@ export const Pin = (props: PinProps) => {
...
     const pinViewOptions: google.maps.marker.PinElementOptions = {
       ...props
     };
 
     const pinElement = new google.maps.marker.PinElement(pinViewOptions);
 
+    /**
+     * Set glyph to glyph container if children are present (rendered via portal).
+     * If both props.glyph and props.children are present, props.children takes priority.
+     *  */
```
[the remark this answers is not in the document]
Comment (benschlegel) [reply src/components/pin.tsx]: fixed in 0aa04d8

File: examples/markers-and-infowindows/src/app.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (benschlegel) [reply examples/markers-and-infowindows/src/app.tsx]: I'm not sure if you have a preference for where you want the svg file locally, I put it in an "assets" folder inside the example now in f4d79a0

**Pull request merged**